File: library/src/main/java/me/shouheng/compress/naming/DefaultNameFactory.java
package me.shouheng.compress.naming;

public final class DefaultNameFactory implements CacheNameFactory {

    public static DefaultNameFactory getFactory() {
        return new DefaultNameFactory();
    }

    private DefaultNameFactory() {
    }

    @Override
    public String getFileName() {
        return System.currentTimeMillis() + (int) (Math.random() * 1000) + ".jpg";
    }

}

File: library/src/main/java/me/shouheng/compress/naming/DefaultNameFactory.kt
package me.shouheng.compress.naming

import android.graphics.Bitmap
import java.util.Random

/** Default name factory implementation. */
object DefaultNameFactory : CacheNameFactory {

    override fun getFileName(format: Bitmap.CompressFormat): String {
        return (System.currentTimeMillis() + Random().nextInt()).toString() + when(format) {
            Bitmap.CompressFormat.PNG -> ".png"
            Bitmap.CompressFormat.JPEG -> ".jpeg"
            Bitmap.CompressFormat.WEBP -> ".webp"
        }
    }
}
